app/models/donation.rb

```diff
@@ -0,0 +1,92 @@
+class Donation
+  include ActiveModel::Model
+
+  attr_accessor :email, :employer, :occupation, :stripe_token, :recurring,
+    :utm_source, :utm_medium, :utm_campaign, :source_url, :amount_in_cents
+
+  validates :email, presence: true,
+    email_format: { message: 'is not a valid email' }
+  validates :employer, presence: true
+  validates :occupation, presence: true
+  validates :stripe_token, presence: true
+  validates :amount_in_cents, presence: true,
+    numericality: { greater_than: 0, only_integer: true }
+
+  def process!
+    if valid?
+      process_payment
+      record_donation
+      create_donate_action
+    else
+      false
+    end
+
```

Comment:
Yeah, I was wondering about that. After thinking about it, though, I kinda thought it should go in the app directory. My thinking is this: everything in lib seems like it should be completely ignorant of the specifics of our app. Ideally, that code should be able to be dropped into any other project that needs to interface with the same external services. By that logic, the stripe gem takes the place of `Integration::Stripe`. What we need is an adapter class that knows just enough about our app and just enough about Stripe to make the two work together. I also want to write a NationBuilder adapter class that I was thinking would live in the app directory. I'm definitely open to your thoughts on this and will go with what you think is best.

As you can probably tell, I've been doing a lot of studying of this kind of stuff and I'm trying to apply the ideas that I'm learning to my work. Everything I'm reading is just suggestions for how to write maintainable code, not rules that must be followed. Any time you have time to engage me in discussion of pros and cons of various strategies would be much appreciated